src/shared/install.c

```diff
@@ -68,6 +68,16 @@ typedef struct {
 
 static int unit_file_lookup_state(UnitFileScope scope, const LookupPaths *paths, const char *name, UnitFileState *ret);
 
+bool unit_type_no_alias(UnitType type) {
+        return IN_SET(type,
+                      UNIT_BUSNAME,
+                      UNIT_MOUNT,
+                      UNIT_AUTOMOUNT,
+                      UNIT_SWAP,
+                      UNIT_SLICE,
+                      UNIT_SCOPE);
+}
```

Comment:
I think whether to whitelist or to blacklist is a question that is orthogonal to the question whether to return true if aliasing is allowed and false otherwise, or the opposite. 

Regardless whether we whitelist or blacklist I am pretty sure we should try to avoid "negated" booleans, and hence call this "unit_type_may_alias()", rather than "unit_type_no_alias()"...

And yeah, I'd vaguely vote for blacklisting too.